prqlc/prql-compiler/src/semantic/resolver/transforms.rs

```diff
@@ -482,131 +553,140 @@ fn into_literal_range(range: (Expr, Expr)) -> Result<(Option<i64>, Option<i64>)>
-    dummy.lineage = Some(val_lineage);
-
-    let pipeline = Expr::new(ExprKind::FuncCall(FuncCall::new_simple(
-        pipeline,
-        vec![dummy],
-    )));
-
-    let env = Module::singleton(param_name, Decl::from(DeclKind::Column(param_id)));
-    resolver.root_mod.module.stack_push(NS_PARAM, env);
-
-    let pipeline = resolver.fold_expr(pipeline)?;
-
-    resolver.root_mod.module.stack_pop(NS_PARAM).unwrap();
-
-    // now, we need wrap the result into a closure and replace
-    // the dummy node with closure's parameter.
-
-    // extract reference to the dummy node
-    // let mut tbl_node = extract_ref_to_first(&mut pipeline);
-    // *tbl_node = Expr::new(ExprKind::Ident("x".to_string()));
-
-    let pipeline = Expr::new(ExprKind::Func(Box::new(Func {
-        name_hint: None,
-        body: Box::new(pipeline),
-        return_ty: None,
-
-        args: vec![],
-        params: vec![FuncParam {
-            name: param_id.to_string(),
-            ty: None,
-            default_value: None,
-        }],
-        named_params: vec![],
-
-        env: Default::default(),
-    })));
-    Ok(pipeline)
+impl Resolver<'_> {
+    /// Simulate evaluation of the inner pipeline of group or window
+    // Creates a dummy node that acts as value that pipeline can be resolved upon.
+    fn fold_by_simulating_eval(
+        &mut self,
+        pipeline: Expr,
+        val: &Expr,
+    ) -> Result<Expr, anyhow::Error> {
+        log::debug!("fold by simulating evaluation");
+        let span = pipeline.span;
+
+        let param_name = "_tbl";
+        let param_id = self.id.gen();
+
+        // resolver will not resolve a function call if any arguments are missing
+        // but would instead return a closure to be resolved later.
+        // because the pipeline of group is a function that takes a table chunk
+        // and applies the transforms to it, it would not get resolved.
+        // thats why we trick the resolver with a dummy node that acts as table
+        // chunk and instruct resolver to apply the transform on that.
+
+        let mut dummy = Expr::new(ExprKind::Ident(Ident::from_name(param_name)));
+        dummy.lineage = val.lineage.clone();
+        dummy.ty = val.ty.clone();
+
+        let pipeline = Expr::new(ExprKind::FuncCall(FuncCall::new_simple(
+            pipeline,
+            vec![dummy],
+        )));
+
+        let env = Module::singleton(param_name, Decl::from(DeclKind::Column(param_id)));
+        self.root_mod.module.stack_push(NS_PARAM, env);
+
+        let mut pipeline = self.fold_expr(pipeline)?;
+
+        self.root_mod.module.stack_pop(NS_PARAM).unwrap();
+
+        // now, we need wrap the result into a closure and replace
+        // the dummy node with closure's parameter.
+
+        // extract reference to the dummy node
+        // let mut tbl_node = extract_ref_to_first(&mut pipeline);
+        // *tbl_node = Expr::new(ExprKind::Ident("x".to_string()));
```

Comment:
```suggestion
```
?

Reply:
I don't even remember why this was commented out or what it was supposed to do... It should go.